includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -708,6 +708,14 @@ private function process_payment_with_deferred_intent( int $order_id ) {
 			if ( 'requires_action' === $payment_intent->status ) {
 				if ( isset( $payment_intent->next_action->type ) && 'redirect_to_url' === $payment_intent->next_action->type && ! empty( $payment_intent->next_action->redirect_to_url->url ) ) {
 					$redirect = $payment_intent->next_action->redirect_to_url->url;
+				} else if ( isset( $payment_intent->next_action->type ) && in_array( $payment_intent->next_action->type, [ 'boleto_display_details', 'oxxo_display_details' ] ) ) {
```

Comment:
Something I'm a little unsure about is when we create the intent for a voucher payment method we're calling the same `create_and_confirm_payment_intent()` method that we use all of our other payment method types which creates and confirms the payment intent in the same request.

When a voucher payment intent is confirmed, Stripe sends the `payment_intent.requires_action` webhook event which we catch and put the order on-hold ([see code](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/class-wc-stripe-webhook-handler.php#L869-L870))

Then on the checkout, we're confirming the payment intent/secret again to display the voucher to the customer.

This doesn't pose any issues from my testing, however, the potential issue I'm concerned about is what happens when something goes wrong with displaying the voucher to the customer?

Because it's already created and confirmed on server side, the webhook has already been handled and the order has been updated from pending payment to on-hold, even though the customer hasn't been shown the voucher code/document to pay.

One change I can make is to update `create_and_confirm_payment_intent()` to not immediately confirm voucher payment method types and let that happen on the front-end, but then that feels a little odd to have a function with the name saying it will confirm the intent but doesn't in some cases.

I'll explore this change above but also open to other suggestions :)

Reply:
I've just pushed up da8b86edc63824320cd4bb0872bd43605a81bb23 & 2f59a28713851076642f642dbe932ee3c1191c3c to address this.

With these latest changes, we will no longer confirm the intent immediately when using voucher payments, which puts the order on-hold. And only confirm the voucher payment during the same request which displays the voucher to the customer.